raft/raft_test.go

```diff
@@ -902,61 +869,55 @@ func TestDisruptiveFollower2AA(t *testing.T) {
 	}
 }
 
-// When the leader receives a heartbeat tick, it should
-// send a MessageType_MsgHeartbeat with m.Index = 0, m.LogTerm=0 and empty entries.
-func TestBcastBeat2AB(t *testing.T) {
-	offset := uint64(1000)
-	// make a state machine with log.offset = 1000
-	s := pb.Snapshot{
-		Metadata: &pb.SnapshotMetadata{
-			Index:     offset,
-			Term:      1,
-			ConfState: &pb.ConfState{Nodes: []uint64{1, 2, 3}},
-		},
-	}
-	storage := NewMemoryStorage()
-	storage.ApplySnapshot(s)
-	sm := newTestRaft(1, nil, 10, 1, storage)
-	sm.Term = 1
-
-	sm.becomeCandidate()
-	sm.becomeLeader()
-	sm.Step(pb.Message{MsgType: pb.MessageType_MsgPropose, Entries: []*pb.Entry{{}}})
-	sm.readMessages() // clear message
-	// slow follower
-	sm.Prs[2].Match, sm.Prs[2].Next = 5, 6
-	// normal follower
-	sm.Prs[3].Match, sm.Prs[3].Next = sm.RaftLog.LastIndex(), sm.RaftLog.LastIndex()+1
-
-	sm.Step(pb.Message{MsgType: pb.MessageType_MsgBeat})
-	msgs := sm.readMessages()
-	if len(msgs) != 2 {
-		t.Fatalf("len(msgs) = %v, want 2", len(msgs))
-	}
-	wantCommitMap := map[uint64]uint64{
-		2: min(sm.RaftLog.committed, sm.Prs[2].Match),
-		3: min(sm.RaftLog.committed, sm.Prs[3].Match),
+func TestHeartbeatUpdateCommit2AB(t *testing.T) {
+	tests := []struct {
+		failCnt    int
+		successCnt int
+	}{
+		{1, 1},
+		{5, 3},
+		{10, 10},
```

Comment:
```suggestion
		{5, 10},
```
Test case of `failCnt ` > `successCnt `